sodiumpp/sodiumpp.cpp

```diff
@@ -355,6 +355,7 @@ std::string sodiumpp::crypto_stream_xor(const std::string &m,const std::string &
 std::string sodiumpp::bin2hex(const std::string& bytes) {
     std::string hex(bytes.size()*2 + 1, 0);
     sodium_bin2hex(&hex[0], hex.size(), reinterpret_cast<const unsigned char *>(&bytes[0]), bytes.size());
+    hex.erase(hex.end() - 1);
     return hex;
```

Comment:
It was an error, looking at sodium_bin2hex doc
https://github.com/jedisct1/libsodium-doc/blob/master/helpers/README.md
"The string is stored into hex and includes a nul byte (\0) terminator."

So the allocation is fine,  but the resulting string would be returned in form like 3 charcters: "ff\0" (char 'f', char 'f', char with value 0) which probably does not make sense for std::string.

It was causing errors?